nextcord/client.py

```diff
@@ -2102,33 +2117,170 @@ def get_application_command(self, command_id: int) -> Optional[BaseApplicationCo
+        search_locales: :class:`bool`
+            Whether to also search through the command's name locales. Defaults to ``False``.
 
         Returns
         -------
-        command: Optional[:class:`BaseApplicationCommand`]
-            Application Command with the given signature. If no command with that signature is
-            found, returns ``None`` instead.
+        command: Optional[:class:`BaseApplicationCommand`, :class:`SlashApplicationSubcommand`]
+            Application Command with the given signature. If no command with that signature is found, ``None`` is returned instead.
         """
-        if isinstance(cmd_type, ApplicationCommandType):
-            actual_type = cmd_type.value
-        else:
-            actual_type = cmd_type
-
         return self._connection.get_application_command_from_signature(
-            name=name, cmd_type=actual_type, guild_id=guild_id
+            type=type.value if isinstance(type, ApplicationCommandType) else type,
+            qualified_name=qualified_name,
+            guild_id=guild_id,
+            search_locales=search_locales,
+        )
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        *,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        qualified_name: str,
+        type: Union[int, ApplicationCommandType],
+        guild_id: Optional[int],
+        *,
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        *,
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @copy_doc(__actual_get_application_command_from_signature)
+    def get_application_command_from_signature(
+        self, *args: Any, search_locales: bool = False, **kwargs: Any
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        REQUIRED_ARGS = ("qualified_name", "type", "guild_id")
+        FUNC = "get_application_command_from_signature()"
+        ARGS_DEPRECATED_MESSAGE = "The signature passed to {name} is deprecated, please use keyword arguments only instead. Converted signature: {converted_signature}."
+        KWARG_DEPRECATION_MESSAGE = 'The keyword argument "{old}=" used in {name} is deprecated, please use "{new}=" instead.'
+
+        kwargs_convert_table: Dict[str, Any] = {"name": "qualified_name", "cmd_type": "type"}
+        VALID_KWARGS = ("type", "qualified_name", "guild_id", "search_locales")
```

Comment:
Wouldn't `search_locales` get consumed by the explicitly defined keyword only argument?

Reply:
Good catch, thanks.